config/v1/types_infrastructure.go

```diff
@@ -28,6 +28,11 @@ type InfrastructureSpec struct {
 
 // InfrastructureStatus describes the infrastructure the cluster is leveraging.
 type InfrastructureStatus struct {
+	// infrastructureName uniquely identifies a cluster with a human friendly name.
+	// Once set it should not be changed. Must be of max length 27 and must have only
+	// alphanumeric or hyphen characters.
+	InfrastructureName string `json:"infrastructureName"`
```

Comment:
nit: `InfrastructureID` because it is an identifier. but looks like @smarterclayton wanted it this way so feel free to keep it as-is

Reply:
I'm good with either `InfrastructureID` or `InfrastructureName`. To me the difference sound more like in terms of uniqueness (an unique IDentifier) than about the content and readability. So in that sense I'd have it as `InfrastructureID`. But `Name` seems to be compliant with similar use cases in the API, e.g. the `GroupName`.

Reply:
Name in our API is a unique in space (but not time) identifier.  Names are things that refer to entities in other systems.  UID is reserved for unique in time and space things we generate ourselves.  If you can set it, it's not a UID, because someone else could set it to the same value.

Reply:
> Name in our API is a unique in space (but not time) identifier. Names are things that refer to entities in other systems. UID is reserved for unique in time and space things we generate ourselves. If you can set it, it's not a UID, because someone else could set it to the same value.

I'm going to pin this comment in my bookmarks :)